src/test/java/com/codersbay/gerhofer/SquareTest.java

```diff
@@ -0,0 +1,31 @@
+package com.codersbay.gerhofer;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SquareTest {
+    @Test
+    @DisplayName("When we create a shape with negative side length we expect to throw an IllegalArgumentException")
+    public void testInvalidSideLengthException() {
+        assertThrows(IllegalArgumentException.class, () -> {
+            new Square(-2d);
+        });
+    }
+
+    @Test
+    @DisplayName("When we test getArea() with side length 2, expecting it to return 4")
+    public void testGetArea() {
+        Square testSquare = new Square(2);
+        assertEquals(4, testSquare.getArea());
+    }
+
+    @Test
+    @DisplayName("When we test getPerimeter() with side length 2, expeting it to return 8")
```

Comment:
ansich schon mal schön getested - die minimal Anforderung überall erfüllt ✔️ 

aaaber cool wärs wenn wir ein paar mehr zahlenkombinationen probieren würden. hierfür bieten sich wsl parameterisierte tests gut an - 
https://www.baeldung.com/parameterized-tests-junit-5

Grenzwerte testen is meist super - also irgendeine sehr kleine zahl, 0, null, eine sehr große Zahl. 
https://www.baeldung.com/parameterized-tests-junit-5